main.tf

```diff
@@ -71,11 +78,12 @@ locals {
     # because node group supports up to 20 types but launch template does not.
     # See https://docs.aws.amazon.com/eks/latest/APIReference/API_CreateNodegroup.html#API_CreateNodegroup_RequestSyntax
     # Keep sorted so that change in order does not trigger replacement via random_pet
-    instance_types  = sort(var.instance_types)
-    ami_type        = local.launch_template_ami == "" ? var.ami_type : null
-    capacity_type   = var.capacity_type
-    labels          = var.kubernetes_labels == null ? {} : var.kubernetes_labels
-    taints          = var.kubernetes_taints
+    instance_types = sort(var.instance_types)
+    ami_type       = local.launch_template_ami == "" ? var.ami_type : null
+    capacity_type  = var.capacity_type
+    labels         = var.kubernetes_labels == null ? {} : var.kubernetes_labels
+
+    taints          = local.is_windows ? concat(local.windows_taint, var.kubernetes_taints) : var.kubernetes_taints
```

Comment:
This taint should be optional and arguably default to not set. See comment above where the taint is defined.